packages/babel-parser/src/parser/expression.js

```diff
@@ -97,11 +101,12 @@ export default class ExpressionParser extends LValParser {
 
   // Convenience method to parse an Expression only
   getExpression(): N.Expression {
-    let scopeFlags = SCOPE_PROGRAM;
+    let paramFlags = PARAM_;
     if (this.hasPlugin("topLevelAwait") && this.inModule) {
-      scopeFlags |= SCOPE_ASYNC;
+      paramFlags |= PARAM_AWAIT;
     }
-    this.scope.enter(scopeFlags);
+    this.scope.enter(SCOPE_PROGRAM);
+    this.param.enter(paramFlags);
```

Comment:
I don't like this name, since `param` could be anything. What about `this.prodp`, or better `this.prodParam`?

Reply:
Well if we don't require `prodParam` to be as easy to speak as `param`, I am good with `prodParam` which is still shorter than the daunting `this.productionParameter`.